dev/TabView/TabView.cpp

```diff
@@ -208,6 +208,39 @@ void TabView::OnSelectedItemPropertyChanged(const winrt::DependencyPropertyChang
 void TabView::OnTabWidthModePropertyChanged(const winrt::DependencyPropertyChangedEventArgs&)
 {
     UpdateTabWidths();
+
+    // Switch the visual states of all tab items to the correct TabViewWidthMode
+    for (auto item : TabItems())
+    {
+        auto tvi = item.try_as<TabViewItem>();
```

Comment:
>tvi [](start = 13, length = 3)

const

Reply:
I think adding `const` doesn't work, since we are later may be assigning a value.

Reply:
ah your right, you could do this is you wanted strict c++ guidelines adherance

```
auto const tvi =[item]()
{
    if(auto tabViewItem =item.try_as<TabViewItem>())
    {
        return tabViewItem;
    }
    return ContainerFromItem(item).try_as<TabViewItem>();
}();
```

Acutally you should probably take this lambda and make it a private helper since you do this code path in a few different places

---
In reply to: [386699994](https://github.com/microsoft/microsoft-ui-xaml/pull/2016#discussion_r386699994) [](ancestors = 386699994)

Reply:
Unfortunately, creating a helper function with the appropiate return type is something Visual Studio does not seem to like:

![image](https://user-images.githubusercontent.com/16122379/76142377-6d51ca80-606d-11ea-9773-f816d107317a.png)

I will use the lambda in the two functions where we have this pattern since a helper function seems to much trouble right now.

Reply:
You are right it is probably not worth the time investment to resolve this, but the reason you you see the pattern all over the place is that when people write new code they copy what they see, the only way for us to get the codebase in line without a huge overhaul is to tackle new code as it is written.  but if it is causing you issue don't worry about it.

That being said, I'm not exactly sure what is going on with your helper function but it should definitely be possible... as for the lambda you can just include a this pointer in your capture block resulting in :

```
auto const tvi =[item, this]()
{
    if(auto tabViewItem =item.try_as<TabViewItem>())
    {
        return tabViewItem;
    }
    return ContainerFromItem(item).try_as<TabViewItem>();
}();
```

Reply:
I see, maybe it is worth using the lambda now, so we can prevent cargo-culturing of nonoptimal code. I've changed the two situations where I introduced that pattern and switched to the lambda you provided.

If I recall correctly, the problem with the helper function was that we would need to return a winrt::com_ptr<TabViewItem> which seems to be problematic as the "normal" TabViewItem is not a winrt type. Passing something different and casting after calling the helper does not seem that good either in my opinion.